Samples/resource-owner-password-credential/Angular2/openiddict-angular2-server/src/openiddict-angular2/Controllers/ResourceController.cs

```diff
@@ -0,0 +1,31 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using openiddict_angular2.Models;
+using openiddict_angular2.Data;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+
+
+namespace openiddict_angular2.Controllers
+{
+    [Authorize]
+    public class ResourceController : Controller
+    {
+        private ApplicationDbContext _context;
+        private UserManager<ApplicationUser> _userManager;
+
+        public ResourceController(ApplicationDbContext context, UserManager<ApplicationUser> userManager)
+        {
+            _context = context;
+            _userManager = userManager;
+        }
+
+        [Route("api/Resource"), HttpGet]
+        public async Task<IActionResult> Get()
+        {
+            var user = await _userManager.GetUserAsync(User);
```

Comment:
if the user is not authorized, a 401 is sent back to the caller, so either remove the line or return a BadRequest with appropriate message or a NotFound